packages/front-end/services/track.ts

```diff
@@ -77,3 +97,52 @@ export default function track(
 
   jitsu.track(event, trackProps);
 }
+
+export function trackSnapshot(
+  event: "create" | "update" | "delete" | "error",
+  props: TrackSnapshotProps
+): void {
+  track(
+    "Experiment Snapshot: " + event, 
+    cleanSnapshotProps(props)
+  );
+}
+
+export function trackReport(
+  event: "create" | "update" | "refresh" | "error" | "delete",
+  props?: TrackSnapshotProps
+): void {
+  track(
+    "Experiment Report: " + event, 
+    props ? cleanSnapshotProps(props) : {}
+  );
+}
+
+function cleanSnapshotProps(props: TrackSnapshotProps): TrackEventProps {
+  props.experiment = props.experiment ? md5(props.experiment) : "";
+  props.id = props.id ? md5(props.id) : "";
```

Comment:
This mutates the original `props` object.  This can be problematic since in several places you define a shared object and re-use it for multiple buttons.  That could result in these strings getting hashed twice.

Better to do this below when creating trackEventProps:

```ts
const trackEventProps: TrackEventProps = {
    ...props,
    dimension_type: parsedDim.type,
    dimension_id: parsedDim.id,
    // Overwrite `id` and `experiment`
    id: props.id ? md5(props.id) : "",
    experiment: props.experiment ? md5(props.experiment) : "",
  };
```